### ISSUE TITLE
install chaincode error 

### ISSUE THREAD

**Issue by Ren-HK** [hyperledger-labs/minifabric#148]
when I up network.I get error when install chaincode.
```
# Run the chaincode install script on cli container ***********
  non-zero return code
  go: github.com/golang/protobuf@v1.3.1: Get "https://proxy.golang.org/github.com/golang/protobuf/@v/v1.3.1.mod": dial tcp 172.217.24.17:443: i/o timeout
  Error: failed to normalize chaincode path: 'go list' failed with: go: github.com/golang/protobuf@v1.3.1: Get "https://proxy.golang.org/github.com/golang/protobuf/@v/v1.3.1.mod": dial tcp 172.217.160.81:443: i/o timeout: exit status 1
  Error: failed to read chaincode package at 'simple_go_1.0.tar.gz': open simple_go_1.0.tar.gz: no such file or directory
```

**Comment by pkirkinezis** [comment]
Check if you have internet access so it can be downloaded the protobuf package

**Comment by litong01** [comment]
Please add more detailed steps for recreating the error. Other wise, there is not much to go on with.

**Comment by Mavlarn** [comment]
I have downloaded the latest version of Minifab docker image, which is on Jan 27th. When I install a contract in go language, there is still an error about the go proxy.

**Comment by Mavlarn** [comment]
It looks like the GOPROXY env is not set when running installcc docker run.

**Comment by litong01** [comment]
@Mavlarn as I have asked earlier, you need to detail what you have done and in what kind of environment, normally you would need to describe each step from beginning so that ones who like to help can possibly know what you've been through. For this kind of errors, my guess is that you are in an environment, access to google site is restricted. To help with that, you will need to create your own spec.yaml file and specify the go proxy as the default spec.yaml file already has but commented out. If the sample proxy is not working for you, you will need to find a proxy which works for you.

**Comment by Mavlarn** [comment]
I updated with latest shell and docker images, when I run `up`, the default simple chaincode contract of go version can be installed.
But, after I created a new channel, and prepared custom chaincode, when I try to install the new chaincode, there is always and error as this:
```
TASK [Run the chaincode install script on cli container] ***********************
fatal: [minifab]: FAILED! => {"changed": true, "cmd": ["docker", "exec", "d5ec399859_cli", "/vars/run/ccinstall.p1.bc.starlets.com.cn.sh"], "delta": "0:00:31.181495", "end": "2021-01-29 16:45:29.588752", "msg": "non-zero return code", "rc": 1, "start": "2021-01-29 16:44:58.407257", "stderr": "Error: chaincode install failed with status: 500 - failed to invoke backing implementation of 'InstallChaincode': could not build chaincode: docker build failed: docker image build failed: docker build failed: Error returned from build: 1 \"go: github.com/hyperledger/fabric-chaincode-go@v0.0.0-20200424173110-d7076418f212: Get \"https://proxy.golang.org/github.com/hyperledger/fabric-chaincode-go/@v/v0.0.0-20200424173110-d7076418f212.mod\": dial tcp 172.217.160.113:443: i/o timeout\n\"", "stderr_lines": ["Error: chaincode install failed with status: 500 - failed to invoke backing implementation of 'InstallChaincode': could not build chaincode: docker build failed: docker image build failed: docker build failed: Error returned from build: 1 \"go: github.com/hyperledger/fabric-chaincode-go@v0.0.0-20200424173110-d7076418f212: Get \"https://proxy.golang.org/github.com/hyperledger/fabric-chaincode-go/@v/v0.0.0-20200424173110-d7076418f212.mod\": dial tcp 172.217.160.113:443: i/o timeout", "\""], "stdout": "", "stdout_lines": []}

PLAY RECAP *********************************************************************
minifab                    : ok=24   changed=0    unreachable=0    failed=1    skipped=26   rescued=0    ignored=0   
```

And I also tried with the sample [chaincode in fabric-sample](https://github.com/hyperledger/fabric-samples/tree/master/asset-transfer-basic/chaincode-go), it can be installed successfully.

I am not sure if there is any error in my chaincode. I created it in vscode, and installed related extensions. There is no error in vscode.  At least it should compile successfully.

So, is there any way I can check the reason it fails? Thanks.

**Comment by litong01** [comment]
@Mavlarn your chaincode was not built. and most likely your chaincode did not compile. If the chaincode comes with minifabric worked fine, the error should be in your own chaincode, maybe missing dependencies or other issues.

**Comment by Mavlarn** [comment]
Thanks.
I also doubt about that. But I cannot find why. Because there is no error in my local env. If there is something missing in dependencies, there will be error in local.

And during the installing, before it fails, I got the `docker inspect ` for the ccenv container.  And I see the GOPROXY in env is not changed. So I thought maybe the env is not working. I attached the result of docker inspect.

[inspect_ccenv.json.txt](https://github.com/hyperledger-labs/minifabric/files/5895472/inspect_ccenv.json.txt)

**Comment by pkirkinezis** [comment]
> Thanks.
> I also doubt about that. But I cannot find why. Because there is no error in my local env. If there is something missing in dependencies, there will be error in local.
> 
> And during the installing, before it fails, I got the `docker inspect ` for the ccenv container. And I see the GOPROXY in env is not changed. So I thought maybe the env is not working. I attached the result of docker inspect.
> 
> [inspect_ccenv.json.txt](https://github.com/hyperledger-labs/minifabric/files/5895472/inspect_ccenv.json.txt)

Have you checked the dependencies ?
The errors when you are trying to install a chain code will be display during the install command .
Also check if all the dependencies can be downloaded during installation from your first error it seems like chain code cannot find a package .
But this has nothing to do with minifabric.

Please also  install PORTAINER  and check the peer logs / orderer logs containers to see after the failed installation of the chain code .

Also try to install the default chain code coming with  mini fabric and see what happens .

**Comment by Mavlarn** [comment]
After several days working on this, It can work now. At first, reply the question above:
1. About the dependency, I can make sure it's ok.
2. From the docker logs, I cannot see any helpful message, just the error as I posted above.

At first, I also thought it was because of the dependency problems. So I took much time to modify the contract and try again and again.

At last, I tried to deploy from vscode. I setup the environment and wallet, and package and deploy from vscode. During installation, there is another error, same as that in this [post](https://jira.hyperledger.org/browse/FAB-17725) .
```
Error: chaincode install failed with status: 500 - failed to invoke backing implementation of 'InstallChaincode': could not build chaincode: docker build failed: docker image build failed: docker build failed: Error returned from build: 1 "go: inconsistent vendoring in /chaincode/input/src/github.com/hyperledger/fabric-samples/chaincode/abstore/go:
	github.com/hyperledger/fabric-contract-api-go@v1.0.0: is explicitly required in go.mod, but not marked as explicit in vendor/modules.txt
```

My error message is about some other libs, but I think it is the same reason.

It said it is some problem in go1.14. So I changed the version to 1.13. Then the installation was ok. 

And at last, thanks for your help, and providing these great tool.
